### DISCUSSION TITLE
ldap problems with Kimai 1.30.7

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#3849] [On-Premise: Installation & Updates]
I use ldap to authenticate; after upgrading (to 1.30.7), I installed laminas according to the documentation. This is the output message:

```
/kimai2$php80 composer require laminas/laminas-ldap --optimize-autoloader -n
Cannot use laminas/laminas-ldap's latest version 2.17.0 as it requires php ~8.0.0 || ~8.1.0 || ~8.2.0 which is not satisfied by your platform.
./composer.json has been updated
Running composer update laminas/laminas-ldap
Loading composer repositories with package information
Updating dependencies
Lock file operations: 1 install, 0 updates, 0 removals
  - Locking laminas/laminas-ldap (2.13.0)
Writing lock file
Installing dependencies from lock file (including require-dev)
Nothing to install, update or remove
Package sensio/framework-extra-bundle is abandoned, you should avoid using it. Use Symfony instead.
Package symfony/debug is abandoned, you should avoid using it. Use symfony/error-handler instead.
Package symfony/inflector is abandoned, you should avoid using it. Use EnglishInflector from the String component instead.
Package php-cs-fixer/diff is abandoned, you should avoid using it. No replacement was suggested.
Generating optimized autoload files
140 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
> symfony-cmd
sh: symfony-cmd: command not found
Script symfony-cmd handling the auto-scripts event returned with error code 127
Script @auto-scripts was called via post-update-cmd
```
My Synology NAS supports php 8.0 (at least Kimai uses it) so that it seems strange that the latest of version laminas-ldap cannot be used. Maybe, it is a "Synology thing" but if someone has a solution, I would not mind (my ldap config did not work until I switched back to the "old" setting `user_method: setUsername` instead of `user_method: setUserIdentifier` - which strangely worked when I was at Kimai 1.30.5)

**Comment by kevinpapst** [comment]
`setUserIdentifier` is for 2.0

1.30.5 does not know this method, see: https://github.com/kimai/kimai/blob/1.30.5/src/Entity/User.php

Run `php -v` to verify which version your are really using.

**Comment by BeckeBauer** [comment]
It might be caused be config of platform:
```
php -v
PHP 8.0.23 (cli) (built: Dec  6 2022 10:30:21) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.23, Copyright (c) Zend Technologies
```
php8.0 is running, however:
```
composer why-not laminas/laminas-ldap 2.17.0
laminas/laminas-ldap 2.17.0 requires php (~8.0.0 || ~8.1.0 || ~8.2.0 but 7.3 is installed) 
Not finding what you were looking for? Try calling `composer update "laminas/laminas-ldap:2.17.0" --dry-run` to get another view on the problem.
```
it seems that somehow php7.3 is still called (although deinstalled from the system)

```
composer update "laminas/laminas-ldap:2.17.0" --dry-run
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires laminas/laminas-ldap ^2.13 -> satisfiable by laminas/laminas-ldap[2.17.0].
    - laminas/laminas-ldap 2.17.0 requires php ~8.0.0 || ~8.1.0 || ~8.2.0 -> your php version (7.3; overridden via config.platform, actual: 8.0.23) does not satisfy that requirement.
```

composer.json shows
```
    "config": {
        "platform": {
            "php": "7.3"
        },
```

I tried to change composer.json to 8.0 but it caused even more errors; any ideas?

**Reply by kevinpapst** [reply]
I don't even know what the problem is. Does LDAP not work?

**Reply by BeckeBauer** [reply]
I was just confused about the error when installing laminas-ldap, that v2.17 cannot be installed because PHP requirements not being met by the system although php8.0 is running - but laminas v2.13 is working with `user_method: setUsername`
I was probably wrong that `user_method: setUserIdentifier` worked with Kimai 1.30.5. Sorry for the confusion

**Reply by kevinpapst** [reply]
Ok. Got it. That is not an error, but only an information. I'll extend the docs to include the `-q` flag to prevent questions 😁
